app/controllers/v0/sessions_controller.rb

```diff
@@ -27,26 +31,21 @@ def saml_callback
         params[:SAMLResponse], settings: saml_settings
       )
 
-      if @saml_response.is_valid?
-        persist_session_and_user!
-        redirect_to SAML_CONFIG['relay'] + '?token=' + @session.token
+      if @saml_response.is_valid? && persist_session_and_user
+        async_create_evss_account(@current_user)
+        redirect_to SAML_CONFIG['relay'] + '?token=' + @session.token + '&level=' + @session.level.to_s
       else
         redirect_to SAML_CONFIG['relay'] + '?auth=fail'
       end
     end
 
     private
 
-    def persist_session_and_user!
-      @session = Session.new(user_attributes.slice(:uuid))
-      @current_user = User.find(@session.uuid)
-      @current_user = saml_user if @current_user.nil? || up_level?
+    def persist_session_and_user
+      @session = Session.new(user_attributes.slice(:uuid).merge(level: loa_current))
+      @current_user = User.new(user_attributes)
```

Comment:
`User.from_session(session)` could be used here as well if you added `RedisStore.find_or_create` similar to `RedisStore.find_or_build`.